modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -137,194 +204,307 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
     void finalize(InputArrayOfArrays inputs_arr, OutputArrayOfArrays) CV_OVERRIDE
     {
         std::vector<Mat> inputs;
-        inputs_arr.getMatVector(inputs);
+        inputs_arr.getMatVector(inputs);  // TODO(perf): slow on non-CPU targets, like OpenCL
         variableChannels = false;
-        for (int i = 1; i < inputs.size(); ++i)
+        outputChannels = inputs[0].size[1];
+        for (size_t i = 1; i < inputs.size(); ++i)
         {
-            if (inputs[i].size[1] != inputs[0].size[1])
-            {
+            int input_channels = inputs[i].size[1];
+            if (outputChannels != input_channels)
                 variableChannels = true;
```

Comment:
May be it would be more efficient to switch `channelsMode` to `ELTWISE_CHANNNELS_SAME` here? This way we can remove `variableChannels` completely.

Reply:
It is handled in this way in `EltwiseInvoker` below:
```
                if (!self.variableChannels)
                {
                    CV_Assert(srcs[i].size == dst.size);
                }
                else if (self.channelsMode == ELTWISE_CHANNNELS_INPUT_0)
                ...
```

if `self.variableChannels == false` then `channelsMode` is not checked anymore.
So it is efficient.

---

> switch channelsMode to ELTWISE_CHANNNELS_SAME

I would like to avoid changing of input parameters. Lets do not "remove" extra checks for next network reshapes.